src/Sarif.UnitTests/FileRegionsCacheTests.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+using System.Collections.ObjectModel;
+using FluentAssertions;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Sarif.UnitTests
+{
+    public class FileRegionsCacheTests
+    {
+        //                                 0          10          20         30
+        //                                 01234 5 6789012 3 45678901 2 345679012
+        private const string s_testText = "line1\r\n line2\r\n  line3\r\n   line4";
+
+        private static ReadOnlyCollection<Tuple<string, Region, Region>> s_testCases =
+            new ReadOnlyCollection<Tuple<string, Region, Region>>(new Tuple<string, Region, Region>[]
+            {
+                // Regions specified only by start line
+                new Tuple<string, Region, Region>(
+                    "line1",
+                    new Region() { StartLine = 1 }, 
+                    new Region() { StartLine = 1, EndLine = 1, StartColumn = 1, EndColumn = 5, CharOffset = 0, CharLength = 5 })
```

Comment:
Shouldn't it be `EndColumn = 6`? Because the line does include column 5 (the "1" at the end of "line1"), and the length of the region is 5, and we're supposed to have length == endColumn - startColumn.

Reply:
I suggest you define a type rather than using `Tuple`. I had to say to myself, "OK, that's the original region, and that's the fully populated region"... I'd have liked to see
```
new CachedRegion
{
    Snippet = ...,
    OriginalRegion = ...,
    FullyPopulatedRegion = ...
}
```

Reply:
um, yes, so why is the test passing? :) will go check

---
In reply to: [202748121](https://github.com/Microsoft/sarif-sdk/pull/945#discussion_r202748121) [](ancestors = 202748121)